Shopify/dawn
# [Cards and Buttons] 3D "Lift" effect on hover

### PR Summary: 

This PR adds an optional "3D Lift" to Cards and Buttons wherein the element appears to lift off the page on hover. The element also receives a subtle shine effect while it's lifted up off the page.

https://user-images.githubusercontent.com/1202812/226444155-9493eb7e-5b5a-41a4-ac47-fd9206d6b596.mp4

### Why are these changes introduced?

Closes #2451. 

### What approach did you take?

The animations are just a couple CSS transitions: 

- The lift effect is achieved by adding a drop shadow and slight rotation on hover. The element does not actually get any larger, it just appears to due to the combination of the shadow + existing border and zoom transitions on these elements. 
- The shine is achieved by adding a pseudo element on top that includes a gradient background. On hover, this background changes its position and opacity.
- These are all activated by a single global class that's conditionally applied the the `body`. 

### Other considerations

Since the effect is purely aesthetic, it becomes inactive when the visitor has `prefers-reduced-motion` active. 

Known bugs: 

- [ ] Currently, buttons don't get the shine. I couldn't find a great place to add the necessary pseudo class to make this happen and didn't want to make things more complicated by adding more wrappers. I don't really mind not having the shine on buttons, but I'd love a second opinion. 
- [x] The buy button sometimes shows an annoying flash when you hover out.
- [x] I'm currently targeting some very specific selectors, and some relatively generic ones. While testing, we should figure out exactly where this hover state should and should not go into effect. 
- [ ] It's possible to have the card move around and react to the mouse's position on it (like what happens on Apple TV). This would require adding some Javascript. 

### Visual impact on existing themes

This feature is optional, and off by default. When merchants activate it by selecting it from the dropdown in Theme Settings, buttons and product cards across their storefront will receive this new hover effect.

### Testing steps/scenarios

1. Visit the demo store's editor.
2. Visit Theme Settings > Animations
3. Select "3D Lift" from the "Hover Effect" dropdown. 
4. Hover over product cards and buttons, see the new hover effect. 

### Demo links

- [Store](https://mavmrr2y9q3cnghq-45840990230.shopifypreview.com)
- [Editor](https://os2-demo.myshopify.com/admin/themes/139559632918/editor)

### Checklist
